riscv-page-tables/src/page_table.rs

```diff
@@ -685,13 +698,22 @@ impl<'a, T: FirstStagePagingMode> FirstStageMapper<'a, T> {
     fn new(
         owner: &'a FirstStagePageTable<T>,
         vaddr: PageAddr<T::MappedAddressSpace>,
+        page_size: PageSize,
         num_pages: u64,
-    ) -> Self {
-        Self {
+    ) -> Result<Self> {
+        if vaddr
+            .checked_add_pages_with_size(num_pages, page_size)
+            .is_none()
+        {
+            return Err(Error::OutOfMapRange);
+        }
```

Comment:
Thinking about this more, there's a subtle problem with returning an error here -- it leaves the PTEs locked, effectively leaking them. I think it's probably best to do the overflow check in the caller, before we start locking PTEs.